andela/ah-leagueOfLegends
# #159965478  User reset the password via email

## What does this PR do?
```
Allows a user to reset their password by entering their email and a reset
 token is sent to the user's email.
```

## Description of Task to be completed?
```
Implemented a forgot password View that allows a user to enter their email which 
sends a reset token to their respective email. Then the reset password View decodes
 the token and checks if it is valid and update users password in the database.
```

## Tools 
```
- RabbitMQ - to install on Mac run  brew install rabbitmq

```

## How should this be manually tested?

```
- Clone this repo and pip install requirements.txt
- run celery worker using command   celery -A authors worker -l info
- Create a virtual environment, start it and run ./manage.py runserver
- Navigate to postman register, confirm your account by checking your email 
   and clicking the confirmation link
- Go to forgot password url (http://127.0.0.1:8000/api/users/reset_password/), 
      Enter your email and send. A password reset link will be sent to your 
      email with the reset token
- Navigate to (http://127.0.0.1:8000/api/users/reset_password/) Enter your email,
       reset token and your new password.

```

## What are the relevant pivotal tracker stories?

[#159965478](https://www.pivotaltracker.com/story/show/159965478)


## Screenshots if available to support PR.

<img width="808" alt="screen shot 2018-09-10 at 18 36 52" src="https://user-images.githubusercontent.com/39088834/45307783-93ee7e00-b528-11e8-9c34-e20a236ae47c.png">




## Questions:

```
Was this PR helpful?

```
